node/src/database/db_migrations.rs

```diff
@@ -74,353 +158,684 @@ impl DbMigratorReal {
+        let subject = DbMigratorReal::new();
+        let mismatched_schema = 0;
+        let target_version = 5; //not relevant
+        let connection = ConnectionWrapperMock::default()
+            .transaction_result(Err(Error::SqliteSingleThreadedMode)); //hard to find a real-like error for this
 
-    #[test]
-    #[should_panic(
-        expected = "Database version is too low and incompatible with any official version: database corrupted"
-    )]
-    fn validation_check_panics_if_the_given_schema_is_lower_than_any_in_the_list() {
-        let _ = DbMigratorReal::schema_initial_validation_check("0.0.0");
+        let result =
+            subject.migrate_database(mismatched_schema, target_version, Box::new(connection));
+
+        assert_eq!(
+            result,
+            Err("SQLite was compiled or configured for single-threaded use only".to_string())
+        )
     }
 
     #[test]
     fn make_updates_panics_if_the_given_schema_is_of_higher_number_than_the_latest_official() {
-        let ending_digit: char = CURRENT_SCHEMA_VERSION.chars().last().unwrap();
-        let higher_number = ending_digit.to_digit(10).unwrap() + 1;
-        let too_advanced = format!(
-            "{}{}",
-            CURRENT_SCHEMA_VERSION
-                .strip_suffix(ending_digit)
-                .unwrap()
-                .to_string(),
-            higher_number
-        );
+        let last_version = CURRENT_SCHEMA_VERSION;
+        let too_advanced = last_version + 1;
+        let migration_utilities = DBMigrationUtilitiesMock::default();
+        let target_version = 5; //not relevant
         let subject = DbMigratorReal::default();
 
-        let result = subject.make_updates(
-            too_advanced.as_str(),
-            DbMigratorReal::list_of_existing_updates(),
-            Box::new(ConnectionWrapperMock::default()),
-        );
+        let captured_panic = catch_unwind(AssertUnwindSafe(|| {
+            subject.make_updates(
+                too_advanced,
+                target_version,
+                Box::new(migration_utilities),
+                DbMigratorReal::list_of_existing_updates(),
+            )
+        }))
+        .unwrap_err();
 
-        assert_eq!(result,Err(format!("Database claims to be more advanced ({}) than the version {} which is the latest released.",too_advanced,CURRENT_SCHEMA_VERSION)))
+        let panic_message = captured_panic.downcast_ref::<String>().unwrap();
+        assert_eq!(*panic_message,format!("Database claims to be more advanced ({}) than the version {} which is the latest released.",too_advanced,CURRENT_SCHEMA_VERSION))
     }
 
-    #[test]
-    fn version_numeric_transcription_works() {
-        let result = DbMigratorReal::str_version_numeric_transcription("3.21.6");
-
-        assert_eq!(*result, [3, 21, 6])
+    #[derive(Default, Debug)]
+    struct DBMigrationRecordMock {
+        old_version_params: Arc<Mutex<Vec<()>>>,
+        old_version_result: RefCell<Vec<usize>>,
```

Comment:
This could probably be a RefCell<usize>, since you don't accept more than one value.